src/ucs/sys/sys.c

```diff
@@ -898,7 +912,8 @@ ucs_status_t ucs_sysv_alloc(size_t *size, size_t max_size, void **address_p,
 #ifdef SHM_HUGETLB
     if (flags & SHM_HUGETLB) {
         huge_page_size = ucs_get_huge_page_size();
-        if (huge_page_size <= 0) {
+        if ((huge_page_size <= 0) || (ucs_get_huge_pages_count() <= 0) ||
+            !ucs_sys_get_mlock_cap()) {
```

Comment:
minor: pls adjust the debug message to write "SHM_HUGETLB is not supported on the system" because CAP_IPC_LOCK absence is specifically disabling SHM_HUGETLB